pkgs/applications/audio/netease-cloud-music-gtk/default.nix

```diff
@@ -0,0 +1,79 @@
+{ lib
+, stdenv
+, glib
+, gtk3
+, curl
+, dbus
+, openssl
+, gst_all_1
+, pkg-config
+, rustPlatform
+, wrapGAppsHook
+, fetchurl
+, fetchFromGitHub
+, makeDesktopItem
+}:
+let
+  pname = "netease-cloud-music-gtk";
+  version = "1.2.2";
+  cargoSha256 = "sha256-A9wIcESdaJwLY4g/QlOxMU5PBB9wjvIzaXBSqeiRJBM=";
+  src = fetchFromGitHub {
+    owner = "gmg137";
+    repo = "netease-cloud-music-gtk";
+    rev = version;
+    sha256 = "sha256-42MaylfG5LY+TiYHWQMoh9CiVLShKXSBpMrxdWhujow=";
+  };
+  desktopIcon = fetchurl {
+    url = "https://github.com/gmg137/netease-cloud-music-gtk/raw/${src.rev}/icons/netease-cloud-music-gtk.svg";
+    sha256 = "sha256-sxa+sj8KdXgvib2xzrwTuA34as1bsP3XabiyFGUHJsc=";
+  };
+  desktopItem = fetchurl {
+    url = "https://github.com/gmg137/netease-cloud-music-gtk/raw/${src.rev}/netease-cloud-music-gtk.desktop";
+    sha256 = "sha256-YtHBjETvVLH02wwQKR18do0Cc4KpToqF8NkefSBcLBE=";
+  };
+in
+rustPlatform.buildRustPackage rec {
+  inherit pname version src cargoSha256;
+
+  cargoPatches = [ ./cargo-lock.patch ];
+
+  nativeBuildInputs = [
+    glib
+    gtk3
+    dbus
+    pkg-config
+    wrapGAppsHook
+  ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+
+  buildInputs = [
+    glib
+    gtk3
+    curl
+    dbus
+    openssl
+  ] ++ (with gst_all_1; [
+    gstreamer
+    gst-plugins-base
+    gst-plugins-good
+    gst-plugins-bad
+    gst-plugins-ugly
+  ]);
+
+  postPatch = ''
+    mkdir -p $out/share/applications
+    mkdir -p $out/share/icons/hicolor/scalable/apps
+    cp ${desktopItem} $out/share/applications/netease-cloud-music-gtk.desktop
```

Comment:
you should be able to just

```
cp netease-cloud-music-gtk.desktop $out/share/applications/netease-cloud-music-gtk.desktop
```

since the file is part of the repo

Reply:
also you can shorten things with install

```
install -D netease-cloud-music-gtk.desktop $out/share/applications/netease-cloud-music-gtk.desktop
```

automatically creates the target directory for you

Reply:
Thanks, I forgot to use the stuff included in the source code directly.